designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,96 @@
+# Search CLI Tool for NuGet.exe and Dotnet.exe
+
+* Status: **Implementing**
+* Author(s): [Advay Tandon](https://github.com/advay26)
+* Issue: [9704](https://github.com/NuGet/Home/issues/9704) - Search CLI Tool for NuGet.exe and Dotnet.exe
+
+## Problem Background
+
+Search functionality from the command line for NuGet.exe and the .NET Core CLI has been a feature request from the NuGet open source community for some time. Additionally, the NuGet API is already equipped to handle search requests, so an extension of this service to command line users would be both practical to implement and also highly valuable.
+
+## Who are the customers
+
+This targets all NuGet.exe and .NET users that want to be able to search for packages from the command line, rather than using NuGet Gallery or VS PMUI. Also, while `nuget.exe list` allows users to search different package sources for packages with a search query, the results are returned sorted by the package Id, rather than by relevance, as proposed for this Search command. Further, the Search command would provide more information on the packages than List currently does.
```

Comment:
This is great! Another common problem that is addressed by your spec are customers that use `nuget.exe` on a private NuGet server that only supports the NuGet V3 protocol (but not V2). Today, these customers will get the following error if they use the `nuget list` command:

> This version of nuget.exe does not support listing packages from package source 'http://get.xxxxxxxx.com/v3/index.json'.

Your spec gives these customers an escape hatch if the server supports the search protocol. We could improve the error message to say something like:

> This version of nuget.exe does not support listing packages from package source 'http://get.xxxxxxxx.com/v3/index.json'. Consider using the nuget.exe search command instead.

What do you think?